external/kafka-0-10-sql/src/test/scala/org/apache/spark/sql/kafka010/KafkaMicroBatchSourceSuite.scala

```diff
@@ -1444,10 +1444,11 @@ abstract class KafkaSourceSuiteBase extends KafkaSourceTest {
 
     require(testUtils.getLatestOffsets(Set(topic)).size === 5)
 
-    // we intentionally starts from second timestamp
+    // we intentionally starts from second timestamp,
+    // except for partition 4 - it starts from first timestamp
```

Comment:
Otherwise the query will fail. This is also good to test setting timestamp per partition too.